File: hotel-smart-booking-backend/src/main/java/com/example/hotelsmartbookingbackend/service/NotificationService.java
package com.example.hotelsmartbookingbackend.service;

import com.example.hotelsmartbookingbackend.dto.response.NotificationResponse;
import com.example.hotelsmartbookingbackend.entity.User;
import com.example.hotelsmartbookingbackend.enums.Role;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface NotificationService {
    
    void sendNotification(User user, String title, String message, String type, Integer referenceId);
    
    void sendNotificationToRoles(List<Role> roles, String title, String message, String type, Integer referenceId);
    
    Page<NotificationResponse> getNotifications(User user, boolean unreadOnly, Pageable pageable);
    
    void markAsRead(Integer id, User user);
    
    void markAllAsRead(User user);
    
    long getUnreadCount(User user);
}

File: hotel-smart-booking-frontend/src/services/notificationService.js
import axiosInstance from './axiosInstance';

export const getNotifications = async (unreadOnly = false, page = 0, size = 10) => {
  const response = await axiosInstance.get('/notifications', {
    params: { unreadOnly, page, size },
  });
  return response.data;
};

export const markAsRead = async (id) => {
  const response = await axiosInstance.put(`/notifications/${id}/read`);
  return response.data;
};

export const markAllAsRead = async () => {
  const response = await axiosInstance.put('/notifications/read-all');
  return response.data;
};

export const getUnreadCount = async () => {
  const response = await axiosInstance.get('/notifications/unread-count');
  return response.data;
};
